### DISCUSSION TITLE
Installing old package versions from the manifest git history

### DISCUSSION THREAD

**Discussion by leumasme** [ScoopInstaller/Scoop#5488] [Ideas]
From what I can find, when installing a specific package version that is different from the current manifest version, scoop just tries to generate the manifest based on the auto-update configuration of the manifest
https://github.com/ScoopInstaller/Scoop/blob/f93028001fbe5c78cc41f59e3814d2ac8e595724/libexec/scoop-download.ps1#L62-L70
https://github.com/ScoopInstaller/Scoop/blob/f93028001fbe5c78cc41f59e3814d2ac8e595724/lib/manifest.ps1#L132-L154
Considering that scoop buckets are a git repository anyway, why doesn't scoop first try to search the git history of the manifest file for the specified version?  
This of course won't work in various scenarios, such as the version being older than it's scoop manifest, but the current approach also does not work in a lot of cases (often old downloads are linked on from a separate page, or the downloads history is paginated, ...). In case of failure, scoop could simply fall back to the current approach of generating the manifest.

**Comment by rashil2000** [comment]
Tracked here: https://github.com/ScoopInstaller/Scoop/issues/3894
